r/tools/nixlibs.R

```diff
@@ -197,7 +197,9 @@ compile_test_program <- function(code) {
   # Note: if we wanted to check for openssl on macOS, we'd have to set the brew
   # path as a -I directory. But since we (currently) only run this code to
   # determine whether we can download a Linux binary, it's not relevant.
-  runner <- "`R CMD config CXX17` `R CMD config CPPFLAGS` `R CMD config CXX17FLAGS` `R CMD config CXX17STD` -E -xc++"
+  r_exec <- paste(R.home(component = "bin"), "R CMD config", sep="/")
```

Comment:
Can we use `R_CMD_config()` here?

https://github.com/apache/arrow/blob/43c05c56b37daa93e76b94bc3e6952d56d1ea3f2/r/tools/nixlibs.R#L380-L382

Reply:
The original author did not use it.
Maybe because all 4 commands need to run at the same time?
I would not know.

Reply:
@nealrichardson Can we use `R_CMD_config()` here?

Reply:
I don't see why not, and I dug through the PR from last year when this was added and don't see any reason why it has to be this way.